grassroots-backend/src/testing/typetests/cast.test.ts

```diff
@@ -0,0 +1,39 @@
+import * as tstyche from "tstyche";
+import { cast, castWithConversion } from "../../grassroots-shared/cast";
+import { IsNumber, Min } from "class-validator";
+
+class Test {
+  @IsNumber()
+  @Min(0)
+  a!: number;
+
+  constructor(a: number) {
+    this.a = a;
+  }
+}
+
+describe("cast", () => {
+  it("should cast in the trivial case", () => {
+    const testCasted = cast(Test, { a: 2 });
+    tstyche.expect(testCasted).type.toBe<Test>();
+  });
+
+  it("should throw a type error if the properties don't overlap", () => {
+    tstyche
+      .expect(cast(Test, { b: 3 }))
+      .type.toRaiseError("does not exist in type");
```

Comment:
@tdresser Glad to see you found TSTyche useful. I am the author of the library.

To draw your attention, now you can use the `.toBeCallableWith()` matcher instead of expecting errors. This and other ability matchers ship with `tstyche@next`. This is a beta release, but there will be nothing breaking for CLI usage anymore.

The assertion would read like this:

```ts
expect(cast).type.not.toBeCallableWith(Test, { b: 3 });
```

There is no error, so you can move it back to the functional test file.

---

Removing `.not` gives this error:

<img width="726" alt="Screenshot 2025-05-05 at 07 45 21" src="https://github.com/user-attachments/assets/6773bbf8-aae6-4fed-8722-a13e692625e7" />